homeassistant/components/daikin/__init__.py

```diff
@@ -43,19 +39,32 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if entry.unique_id is None or ".local" in entry.unique_id:
         hass.config_entries.async_update_entry(entry, unique_id=conf[KEY_MAC])
 
-    daikin_api = await daikin_api_setup(
-        hass,
-        conf[CONF_HOST],
-        conf.get(CONF_API_KEY),
-        conf.get(CONF_UUID),
-        conf.get(CONF_PASSWORD),
-    )
-    if not daikin_api:
-        return False
+    session = async_get_clientsession(hass)
+    host = conf[CONF_HOST]
+    try:
+        async with asyncio.timeout(TIMEOUT):
+            device: Appliance = await DaikinFactory(
+                host,
+                session,
+                key=entry.data.get(CONF_API_KEY),
+                uuid=entry.data.get(CONF_UUID),
+                password=entry.data.get(CONF_PASSWORD),
+            )
+        _LOGGER.debug("Connection to %s successful", host)
+    except TimeoutError as err:
+        _LOGGER.debug("Connection to %s timed out", host)
```

Comment:
might be nice to tell them how long the timeout was in case we find out its too short later

Reply:
Can we get that value from the error?

Reply:
Isn't in above in the `asyncio.timeout()` block?

Reply:
Oh lol, never mind, was reading this on mobile